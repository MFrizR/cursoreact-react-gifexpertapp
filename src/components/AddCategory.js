import React, { useState } from 'react';
import PropsTypes from 'prop-types';

const AddCategory = ({ setCategories } ) => {

    const [inputValue, setInputValue] = useState(' ');

    const handleInputChange = (e) => {
        console.log(e.target.value);
        setInputValue (e.target.value)
    }

    const handleSubmit = (e) => {
        // Prevenir comportamiento por defecto del formulario
        e.preventDefault();
        

        if(inputValue.trim().length > 2){
            setCategories( cats => [  inputValue,...cats]);
            setInputValue('');
        }
        console.log('Submit Hecho');
    }

    return (
        <>
        <form onSubmit = { handleSubmit }>
            <input
                type= "text"
                value= { inputValue }
                onChange = { handleInputChange }
            />
        </form>
          

          
        </>
    )
}

AddCategory.propsTypes = {
    setCategories: PropsTypes.func.isRequired
};

 AddCategory.defaultProps = {

    //setCategories: 
};  

export default AddCategory
