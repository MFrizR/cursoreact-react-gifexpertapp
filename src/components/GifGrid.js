//import React, { useState, useEffect } from 'react'
import React from 'react';
import PropsTypes from 'prop-types'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from './helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {


    const {data, loading} = useFetchGifs( category );  

//console.log(data)
//const [count, setCount] = useState(0);

/* const [images, setImages] = useState([]);


 */


//getGifs();


    return (
        <>
        <h3>{ category }</h3>
         { loading && <p>Loading</p> } 
        <div className="card-grid">
           

            {
                data.map (( img ) => (

                    <GifGridItem 
                        { ...img }
                        key = { img.id }
                    />
                ))
            }

          {/*   <ol>
                {
                    images.map( img => (
                        <li key={ img.id }>{ img.title }</li>
                    ))
                }
            </ol> */}
            {/* <h3>{ count }</h3> */}
            {/* <button onClick={ () => setCount( count +1 )}></button> */}
         </div>
        </>
    )
}

GifGrid.PropsTypes = {

    category: PropsTypes.string.isRequired
}
