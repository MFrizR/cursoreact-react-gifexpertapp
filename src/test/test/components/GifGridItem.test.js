import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../../components/GifGridItem"


describe('pruebas en <GifGridItem />', () => {
    
    const title = 'un titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title= {title} url={url}/>)

    test('1.- Debe de mostar el componente correctamente ', () => {
        
        
        expect ( wrapper ).toMatchSnapshot();
    })

    test('2.- Debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );
    })
    
    test('3.- Debe de obtener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );
        
    })

    test('4.- Debe de tener card', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('card')).toBe( true );
        
    })
    
    
})
