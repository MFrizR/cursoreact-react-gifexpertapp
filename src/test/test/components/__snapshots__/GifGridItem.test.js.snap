// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> 1.- Debe de mostar el componente correctamente  1`] = `
<div
  className="card"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
